governance-aepp/src/utils/wallet.js

```diff
@@ -15,48 +15,47 @@ export const wallet = {
   walletName: null,
 
   async disconnect () {
-    await this.client.disconnectWallet()
-    this.walletName = null
-    this.pub = null
-    this.balance = null
-    await this.scanForWallets()
+    await this.client.disconnectWallet();
+    this.walletName = null;
+    this.pub = null;
+    this.balance = null;
+    await this.scanForWallets();
   },
+
   async getReverseWindow () {
-    const iframe = document.createElement('iframe')
-    //iframe.src = 'https://base.aepps.com/'
-    iframe.src = 'https://localhost:8080/'
-    iframe.style.display = 'none'
-    document.body.appendChild(iframe)
-    return iframe.contentWindow
+    const iframe = document.createElement('iframe');
+    iframe.src = 'https://base.aepps.com/';
+    //iframe.src = 'https://localhost:8080/';
+    iframe.style.display = 'none';
+    document.body.appendChild(iframe);
+    return iframe.contentWindow;
   },
-  async scanForWallets () {
+  async scanForWallets (successCallback) {
     const scannerConnection = await BrowserWindowMessageConnection({
-      connectionInfo: { id: 'spy' }
-    })
-    const detector = await Detector({ connection: scannerConnection })
+      connectionInfo: { id: 'spy' },
+    });
+    const detector = await Detector({ connection: scannerConnection });
     const handleWallets = async function ({ wallets, newWallet }) {
-      if (confirm(`Do you want to connect to wallet ${newWallet.name}`)) {
-        detector.stopScan()
-        await this.client.connectToWallet(await newWallet.getConnection())
-        await this.client.subscribeAddress('subscribe', 'current')
-        this.pub = await this.client.address()
-        this.balance = await this.client.balance(this.pub)
-        this.walletName = this.client.rpcClient.info.name
-      }
-    }
-    detector.scan(handleWallets.bind(this))
+      detector.stopScan();
+      await this.client.connectToWallet(await newWallet.getConnection());
+      await this.client.subscribeAddress('subscribe', 'current');
+      aeternity.client = this.client;
+      await aeternity.initProvider();
+      successCallback();
+    };
+
+    detector.scan(handleWallets.bind(this));
   },
-  async init () {
+  async init (successCallback) {
     // Open iframe with Wallet if run in top window
-    window !== window.parent || await this.getReverseWindow()
-    //
+    window !== window.parent || await this.getReverseWindow();
```

Comment:
I think we should not include the reverse window iFrame if the integration is broken anyways